File: types/ConfigSchema.ts
import { Cfgu } from './Cfgu';

export interface ConfigSchemaContentsValue extends Cfgu {};
export type ConfigSchemaContents = { [ConfigKey: string]: ConfigSchemaContentsValue };

/**
 * A file containing binding records linking each unique `ConfigKey` to its corresponding `Cfgu` declaration.
 * https://configu.com/docs/config-schema/
 */
export interface ConfigSchema {
  name: string;
  contents: ConfigSchemaContents;
}

File: java/src/entities/ConfigSchema.java
package entities;

import com.fasterxml.jackson.annotation.*;

/**
 * An interface of a <uid>.cfgu.[json|yaml] file, aka ConfigSchema
 * that contains binding records between a unique Config <key> and its Cfgu declaration
 */
@lombok.Data
public class ConfigSchema {
    @lombok.Getter(onMethod_ = {@JsonProperty("contents")})
    @lombok.Setter(onMethod_ = {@JsonProperty("contents")})
    private String contents;
    @lombok.Getter(onMethod_ = {@JsonProperty("path")})
    @lombok.Setter(onMethod_ = {@JsonProperty("path")})
    private String path;
    @lombok.Getter(onMethod_ = {@JsonProperty("type")})
    @lombok.Setter(onMethod_ = {@JsonProperty("type")})
    private ConfigSchemaType type;
    @lombok.Getter(onMethod_ = {@JsonProperty("uid")})
    @lombok.Setter(onMethod_ = {@JsonProperty("uid")})
    private String uid;
}
